profiles/serializers_test.py

```diff
@@ -304,3 +309,134 @@ def test_delete_employment(self):
 
         # Other profile is unaffected
         assert employment3.profile.work_history.count() == 1
+
+
+class ProfileFilledOutTests(ESTestCase):
+    """Tests for validating filled out profiles"""
+
+    @classmethod
+    def setUpTestData(cls):
+        """
+        Create a profile and social auth
+        """
+        with mute_signals(post_save):
+            cls.profile = ProfileFactory.create(
+                filled_out=True,
+                agreed_to_terms_of_service=True,
+            )
+        EmploymentFactory.create(profile=cls.profile)
+        EducationFactory.create(profile=cls.profile)
+        cls.profile.user.social_auth.create(
+            provider=EdxOrgOAuth2.name,
+            uid="{}_edx".format(cls.profile.user.username)
+        )
+
+    def setUp(self):
+        """
+        Create a profile and social auth
+        """
+        serializer = ProfileFilledOutSerializer(self.profile)
+        self.data = serializer.to_representation(self.profile)
+
+    def test_success(self):
+        """
+        Test a successful validation
+        """
+        ProfileFilledOutSerializer(data=self.data).is_valid(raise_exception=True)
+
+    def test_filled_out_false(self):
+        """
+        filled_out cannot be set to false
+        """
+        self.data['filled_out'] = False
+        with self.assertRaises(ValidationError) as ex:
+            ProfileFilledOutSerializer(data=self.data).is_valid(raise_exception=True)
+        assert ex.exception.args[0] == {'non_field_errors': ['filled_out cannot be set to false']}
+
+    def test_agreed_to_terms_of_service_false(self):
+        """
+        agreed_to_terms_of_service cannot be set to false
+        """
+        self.data['agreed_to_terms_of_service'] = False
+        with self.assertRaises(ValidationError) as ex:
+            ProfileFilledOutSerializer(data=self.data).is_valid(raise_exception=True)
+        assert ex.exception.args[0] == {'non_field_errors': ['agreed_to_terms_of_service cannot be set to false']}
+
+    def test_required_fields(self):
+        """
+        All fields are required after a profile is marked filled_out
+        """
+        # skip fields that are generated, and work_history and education which are tested elsewhere
+        keys = set(ProfileFilledOutSerializer.Meta.fields) - {
+            'username',
+            'profile_url_full',
+            'profile_url_large',
+            'profile_url_medium',
+            'profile_url_small',
+            'pretty_printed_student_id',
+            'work_history',
+            'education',
+        }
```

Comment:
this set of readonly field names seems to be copied in ProfilePATCHTests.test_patch_own_profile. we should maintain this in some common class. does it make sense to define it in ProfileSerializer or ProfileFilledOutSerializer?